roles/reproducer/tasks/ci_data.yml

```diff
@@ -4,15 +4,29 @@
     - cifmw_job_uri is defined
   delegate_to: localhost
   block:
+    - name: Parse URL
```

Comment:
wondering if this may be moved as a block var instead of creating that "local" fact?
Something like:
```YAML
- name: Run block only if we want to re-run a job
  when:
    - cifmw_job_uri is defined
  vars:
    uri_regex_pattern: "....."
    matched_url: "{{ cifmw_job_uri | regex_search(uri_regex_pattern) }}"
  block:
....
```

Not sure if the `vars` would be interpreted (and thus fail if the job uri isn't set) though.

Reply:
Nice, much cleaner, should we as best practice avoid using `set_fact` because it's like setting a global var?